src/Compilers/CSharp/Test/Semantic/Semantics/LocalFunctionTests.cs

```diff
@@ -77,6 +77,24 @@ private void VerifyDiagnostics(string source, CSharpCompilationOptions options,
         }
 
         [Fact]
+        public void VarLocalFunction()
+        {
+            var src = @"
+class C
+{
+    void M()
+    {
+        var local() => 0;
+        int x = local();
+   } 
+}";
+            VerifyDiagnostics(src,
+                // (6,9): error CS0825: The contextual keyword 'var' may only appear within a local variable declaration or in script code
+                //         var local() => 0;
+                Diagnostic(ErrorCode.ERR_TypeVarNotFound, "var").WithLocation(6, 9));
+        }
+
+        [Fact]
```

Comment:
Also what hapens if `var` is an identifier in the function.  What error do we produce?